client/src/app/models/task.ts

```diff
@@ -0,0 +1,20 @@
+export interface Task {
+    _id: string;
+    title: string;
+    description: string;
+    createdAt: Date;
+    updatedAt: Date;
+    tasks:[string];
+}
+
+export interface TaskUpdate{
+    _id:string;
+    title: string;
+    description:string;
+}
+export interface TaskAdd{
+    title: string;
+    description:string;
+    createdBy:string;
+    parentId:string|null;
```

Comment:
you could use **?** operator to make this parameter optional
**parentId?:string**
this means that we can omit parentId parameter

Reply:
Changed